src/compile/legend/parse.ts

```diff
@@ -143,21 +143,27 @@ function getProperty<K extends keyof LegendComponentProps>(
     //   return getFirstDefined(specifiedLegend.clipHeight, properties.clipHeight(properties.type(...)));
 
     case 'direction':
-      return direction({legend, legendConfig, timeUnit, channel, scaleType});
+      return direction({
+        legend,
+        legendConfig,
+        timeUnit,
+        channel,
+        scaleType
+      }) as LegendComponentProps[K];
 
     case 'format':
       // We don't include temporal field here as we apply format in encode block
       if (isTimeFormatFieldDef(fieldDef)) {
         return undefined;
       }
-      return numberFormat(fieldDef, legend.format, model.config);
+      return numberFormat(fieldDef, legend.format, model.config) as any;
```

Comment:
Why is `direction` the only one that can use `LegendComponentProps[K]`?

Reply:
We can do as any or as key of props. I don't think tye latter gives us any benefits so I'd just use any everywhere.

Reply:
Yep, in general we better avoid `any` since it normally means "I'm giving up on TS for this part.